WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -15,21 +15,25 @@ class JetpackWindowManager: WindowManager {
     }
 
     override func showUI(for blog: Blog?) {
-        // If the user is logged in and has blogs sync'd to their account
-        if AccountHelper.isLoggedIn && AccountHelper.hasBlogs {
-            showAppUI(for: blog)
+        if AccountHelper.isLoggedIn {
+            if AccountHelper.hasBlogs {
+                // If the user is logged in and has blogs sync'd to their account
+                showAppUI(for: blog)
+            } else {
+                // If the user doesn't have any blogs, but they're still logged in, log them out
+                // the `logOutDefaultWordPressComAccount` method will trigger the `showSignInUI` automatically
+                AccountHelper.logOutDefaultWordPressComAccount()
```

Comment:
I need to double-check that this works like I think it should and doesn't carry a logic error from before.

Reply:
It looks like it's OK based on the version before CMF: https://github.com/wordpress-mobile/WordPress-iOS/commit/37e0f2035acdbd887473b708a266b99468a4b33c

Reply:
Thanks for working on this @twstokes 🙇 

Since the whole `showUI` method was revamped, I tested the sign in and sign out and it worked, and the content import event is no longer tracked. But I couldn't test the case where the user is signed in but doesn't have blogs:

```
AccountHelper.loggedIn && !AccountHelper.hasBlogs
```

In my opinion, It would be great to fix this with minimal changes to be extra confident that we're not breaking existing behavior. I tested the following code and it fixed the event tracking issue. Let me know what do you think:

<img width="853" alt="CleanShot 2022-12-05 at 22 56 09@2x" src="https://user-images.githubusercontent.com/9609223/205752146-94bf96f5-eda4-46aa-94a1-7a928f752e6d.png">

This is not blocking so I'll approve the PR and I'll let you decide whether you want to make the changes.

Reply:
Thanks @salimbraksa!

> It would be great to fix this with minimum changes to be extra confident that we're not breaking existing behavior

I agree with this, while at the same time also think we need to be confident in how to test the existing behavior. I'll work on testing steps we can trust for the case you pointed out (e.g. they're logged in but don't have blogs). 🙇 If I don't have confidence in that test I'll be glad to revert things back to a simpler solution. 👍 

I'd also feel better if none of the migration logic was called, even if `shouldImportMigrationData` returns `false` based on the feature flag - it seems like a cleaner separation.

Reply:
>I'd also feel better if none of the migration logic was called, even if shouldImportMigrationData returns false based on the feature flag - it seems like a cleaner separation.

Good point 👍 

>I'll work on testing steps we can trust for the case you pointed out

Sounds good, thanks @twstokes 🙇